File: .sdk/tm/rb/utility/feature_hook.rb
# EvilInsultGenerator SDK utility: feature_hook
module EvilInsultGeneratorUtilities
  FeatureHook = ->(ctx, name) {
    return unless ctx.client
    features = ctx.client.features
    return unless features
    features.each do |f|
      f.send(name, ctx) if f.respond_to?(name)
    end
  }
end

File: php/utility/FeatureHook.php
<?php
declare(strict_types=1);

// EvilInsultGenerator SDK utility: feature_hook

class EvilInsultGeneratorFeatureHook
{
    public static function call(EvilInsultGeneratorContext $ctx, string $name): void
    {
        if (!$ctx->client) {
            return;
        }
        $features = $ctx->client->features ?? null;
        if (!$features) {
            return;
        }
        foreach ($features as $f) {
            if (method_exists($f, $name)) {
                $f->$name($ctx);
            }
        }
    }
}
